src/VueStorefront/PageResult/Navigation/NavigationPageResultHydrator.php

```diff
@@ -31,14 +39,104 @@ public function hydrate(PageLoaderContext $pageLoaderContext, CategoryEntity $ca
     {
         $this->pageResult->setCmsPage($cmsPageEntity);
 
-        $this->setBreadcrumbs($category, $pageLoaderContext->getContext()->getSalesChannel()->getNavigationCategoryId());
+        $this->setBreadcrumbs(
+            $category,
+            $pageLoaderContext->getContext()->getSalesChannel()->getNavigationCategoryId()
+        );
 
         $this->pageResult->setResourceType($pageLoaderContext->getResourceType());
         $this->pageResult->setResourceIdentifier($pageLoaderContext->getResourceIdentifier());
 
+        $this->getAvailableFilters();
+
         return $this->pageResult;
     }
 
+    /**
+     * @TODO: Destruct method into dedicated services and handlers
+     */
+    private function getAvailableFilters(): array
+    {
+        /**
+         * Get all listing slots
+         */
+        $listingSlots = $this->pageResult->getCmsPage()->getSections()->getBlocks()->getSlots()->filter(function($slot) {
```

Comment:
you can simply use `\Shopware\Core\Content\Cms\CmsPageEntity::getElementsOfType`